.github/workflows/_deploy-github-pages.yaml

```diff
@@ -0,0 +1,80 @@
+# Copyright (c) 2023 - 2023, Oracle and/or its affiliates. All rights reserved.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/.
+
+# This workflow deploys the documentations to GitHub Pages.
+name: Deploy static content to Pages
+on:
+  workflow_call:
+    inputs:
+      artifact-name:
+        type: string
+        required: true
+        description: The artifact name that contains docs content
+      artifact-sha256:
+        type: string
+        required: true
+        description: The sha of the artifact that contains docs content
+# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+# Allow only one concurrent deployment, skipping runs queued between the run in-progress and latest queued.
+# However, do NOT cancel in-progress runs as we want to allow these production deployments to complete.
+concurrency:
+  group: pages
+  cancel-in-progress: false
+
+jobs:
+  # Download the generated docs archive, verify its hash, and deploy to GitHub pages.
+  deploy:
+    environment:
+      name: github-pages
+      url: ${{ steps.deployment.outputs.page_url }}
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: Check out repository
+      uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3.5.2
+      with:
+        fetch-depth: 0
+
+    - name: Download artifact
+      uses: actions/download-artifact@9bc31d5ccc31df68ecc42ccf4149144866c47d8a # v3.0.2
+      with:
+        name: ${{ inputs.artifact-name }}
+        path: dist
+
+    # Verify hashes by first computing hashes for the artifacts and then comparing them
+    # against the hashes for the artifact.
+    - name: Verify the artifact hash
+      env:
+        ARTIFACT_HASH: ${{ inputs.artifact-sha256 }}
+      run: |
+        set -euo pipefail
+        echo "Hash of package should be $ARTIFACT_HASH."
+        echo "$ARTIFACT_HASH" | base64 --decode | sha256sum --strict --check --status || exit 1
+
+    # Prepare the docs content.
+    - name: Prepare docs for release
+      env:
+        DIST_PATH: dist
+      run: |
+        DOCS_PATH=$(find "$DIST_PATH" -depth -type f -name 'macaron-*-docs-html.zip' | head -n 1)
+        if [[ -z "${DOCS_PATH}" ]];
+        then
+            echo "Unable to find Macaron docs files in ${DIST_PATH}."
+            exit 1
+        fi
+        unzip "$DOCS_PATH"
+
+    - name: Setup Pages
+      uses: actions/configure-pages@f156874f8191504dae5b037505266ed5dda6c382 # v3.0.6
+    - name: Upload artifact
```

Comment:
I think we should name it `Upload GitHub pages` or `Upload Documentation`.

Reply:
See commit [3f2d6ff](https://github.com/oracle-samples/macaron/pull/249/commits/3f2d6ff8d45c06523bcfcfd2bb3fca5ab015256f).